mne/viz/brain/tests/test_brain.py

```diff
@@ -0,0 +1,154 @@
+                       subjects_dir=subjects_dir)
+
+    with pytest.raises(ValueError):
+        brain_data.add_data(array=np.zeros(3))
+    with pytest.raises(ValueError):
+        brain_data.add_data(array=hemi_data, thresh=0)
+    with pytest.raises(ValueError):
+        brain_data.add_data(array=hemi_data, transparent=0)
+    with pytest.raises(ValueError):
+        brain_data.add_data(array=hemi_data, remove_existing=0)
+    with pytest.raises(ValueError):
+        brain_data.add_data(array=hemi_data, time_label_size=0)
+    with pytest.raises(ValueError):
+        brain_data.add_data(array=hemi_data, scale_factor=0)
+    with pytest.raises(ValueError):
+        brain_data.add_data(array=hemi_data, vector_alpha=0)
+    with pytest.raises(ValueError):
+        brain_data.add_data(array=hemi_data, verbose=0)
+    with pytest.raises(ValueError):
+        brain_data.add_data(hemi_data, min=fmin, hemi=hemi,
+                            max=fmax, vertices=None)
+
+    brain_data.add_data(hemi_data, min=fmin, hemi=hemi, max=fmax,
+                        colormap='hot', vertices=hemi_vertices,
+                        colorbar=False)
+
+    if backend_name != "mayavi":
+        brain_data.show()
+
+
+def test_brain_colormap():
+    """Test brain's colormap functions."""
+    from matplotlib import cm
+    colormap = "coolwarm"
+    alpha = 1.0
+    min = 0.0
+    mid = 0.5
+    max = 1.0
+    center = None
+    _calculate_lut(colormap, alpha=alpha, fmin=min,
+                   fmid=mid, fmax=max, center=center)
+    center = 0.0
+    colormap = cm.get_cmap(colormap)
+    _calculate_lut(colormap, alpha=alpha, fmin=min,
+                   fmid=mid, fmax=max, center=center)
+
+
+def test_brain_time_viewer(backends_3d):
+    """Test of brain's time viewer."""
+    backend_name = get_3d_backend()
+    data_path = sample.data_path()
+    hemi = 'both'
+    surf = 'inflated'
+    subject_id = 'sample'
+    subjects_dir = path.join(data_path, 'subjects')
+    brain = Brain(subject_id, hemi, surf, subjects_dir=subjects_dir)
+
+    with pytest.raises(KeyError):
+        TimeViewer(brain)
+
+    brain.data['time'] = None
+
+    with pytest.raises(ValueError):
+        TimeViewer(brain)
+
+    if backend_name != "ipyvolume":
+        pytest.skip()
+
+    brain.data['time'] = np.zeros(1)
+    brain.data['time_idx'] = np.zeros(1).astype(np.int)
+    brain.data['time_label'] = "0"
+
+    time_viewer = TimeViewer(brain)
+    time_viewer.show()
+
+
+def test_brain_colorbar(backends_3d):
+    """Test of brain's colorbar."""
+    from matplotlib import cm
+
+    backend_name = get_3d_backend()
+    data_path = sample.data_path()
+    hemi = 'both'
+    surf = 'inflated'
+    subject_id = 'sample'
+    subjects_dir = path.join(data_path, 'subjects')
+    brain = Brain(subject_id, hemi, surf, subjects_dir=subjects_dir)
+
+    with pytest.raises(KeyError):
+        ColorBar(brain)
+
+    if backend_name != "ipyvolume":
```

Comment:
use parameterize and skip

Reply:
`parametrize()` is already virtually used by the `backends_3d` fixture. This `skip()` is a temporary solution as I intend to port `ColorBar` and `TimeViewer` to the other backends hopefully through other PRs, I'll communicate on this by a `# XXX` comment if you don't mind.